### PR TITLE
[BEAM-5364] Check if validation is disabled when validating BigtableSource

### PR DESCRIPTION
**Pull request by kevinsi4508**
Check if validation is disabled when validating BigtableSource.
In populateDisplayData, if a runtime parameter is not yet  populated, use a dummy text.

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by kevinsi4508** [comment]
@chamikaramj, could you take a loook?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java [hunk 1]
```diff
@@ -159,13 +159,33 @@ void validate() {
...
+   *     construction" if the runtime parameter is not accessible for debugging purpose.
+   */
+  @Nullable
+  static <T> String getDisplayDataForParameter(@Nullable ValueProvider<T> parameter) {
+    if (parameter == null) return null;
+
+    if (parameter.isAccessible()) {
+      return String.valueOf(parameter.get());
+    }
+    return "Unavailable during pipeline construction";
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java]: This is error-prone. Please raise an error instead of returning a String here.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java [hunk 2]
```diff
@@ -159,13 +159,33 @@ void validate() {
...
+  }
+
   void populateDisplayData(DisplayData.Builder builder) {
     builder
         .addIfNotNull(
-            DisplayData.item("projectId", getProjectId()).withLabel("Bigtable Project Id"))
+            DisplayData.item("projectId", getDisplayDataForParameter(getProjectId()))
+                .withLabel("Bigtable Project Id"))
+        .addIfNotNull(
+            DisplayData.item("instanceId", getDisplayDataForParameter(getInstanceId()))
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java]: I'm not sure why we had to update display-data related logic for this PR (fix validation issue). Should these updates be a separate PR ?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java (hunk 1 above)
Comment (kevinsi4508) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java]: This is expected as at pipeline construction time, runtime parameters don't have values yet and we don't want fail it.

I do think this is a bit misleading because parameter.isAccessible() could mean something else.  Any suggestion what we should do?

zhouyunqing@ can explain better.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java (hunk 2 above)
Comment (kevinsi4508) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java]: I can move this to a separate PR.

Comment (kevinsi4508) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java]: I created a separate PR for disabling validation: https://github.com/apache/beam/pull/6413.

**Comment by kevinsi4508** [comment]
Going to break this PR into 2.
